mpas_analysis/ocean/nino34_timeseries.py

```diff
@@ -0,0 +1,198 @@
+			yLabel, figureName, linewidths=3)
+
+ 
+def compute_nino34_index(regionSST, config):
+    """
+    Computes nino34 index time series.  It follow the standard nino34 algorithm, i.e.,
+
+    1) Compute monthly average SST in the region
+    2) Computes anomalous SST
+    3) Performs a 5 point running mean over the anomalies
+
+    This routine requires regionSST to be the SSTs in the nino3.4 region ONLY.
+    It is defined as lat > -5S and lat < 5N and lon > 190E and lon < 240E.
+
+    Further, regionSST must be a dataArray
+
+    Author: Luke Van Roekel
+    Last Modified: 03/20/2017
+    """
+
+    inDirectory = config.get('input', 'baseDirectory')
+    namelistFileName = config.get('input', 'oceanNamelistFileName')
+    namelist = NameList(namelistFileName, path=inDirectory)
+    calendar = namelist.get('config_calendar_type')
+
+    # Compute monthly average climatology of SST
+    monthlyClimatology = climatology.compute_monthly_climatology(regionSST, calendar)
+
+    anomalySST = regionSST.groupby('month') - monthlyClimatology
+
+    return _running_mean(anomalySST.to_pandas())
+
+
+def compute_nino34_spectra(nino34Index, config):
+    """
+    Computes power spectra of Nino34 index.
+
+    nino34Index is the NINO index computed by compute_nino34_index
+
+    The algorithm follows the NCL cvdp package see
+    http://www.cesm.ucar.edu/working_groups/CVC/cvdp/code.html
+
+    NOTE: this routine requires the signal and stats module from scipy
+
+    Author: Luke Van Roekel
+    Last Modified: 03/21/2017
+    """
+
+    window = signal.tukey(len(nino34Index.values), alpha=0.1)
+    f, Pxx = signal.periodogram(window*nino34Index.values, 1.0 / (86400.*30.))
+
+#   computes power spectra, smoothed with 5 point running mean
+    pxxSmooth = _running_mean(pd.Series(Pxx))
+
+#   compute 99 and 95% confidence intervals and red-noise process
+#   Uses Chi squared test
+
+    r = _autocorr(nino34Index.values)[0, 1]
+    r2 = 2.*r
+    rsq = r**2
+
+#   In the temp2 variable, f is converted to give wavenumber i.e., 0,1,2,...,N/2
+    temp2 = r2*np.cos(2.*np.pi*f*86400.*30.)
+    mkov = 1. / (1. + rsq - temp2)
+
+    sum1 = np.sum(mkov)
+    sum2 = np.sum(pxxSmooth)
+    scale = sum2 / sum1
+
+    xLow = stats.chi2.interval(0.95, 10)[1]/10.
+    xHigh = stats.chi2.interval(0.99, 10)[1]/10.
+
+#   return Spectra, 99% confidence level, 95% confidence level, and Red-noise fit
+    return f, pxxSmooth, mkov*scale*xHigh, mkov*scale*xLow, mkov*scale
+
+
+def _autocorr(x, t=1):
+    """
+    Computes lag one auto-correlation for the NINO34 spectra calculation
+
+    Author: Luke Van Roekel
+    Last Modified: 03/20/2017
+    """
+
+    return np.corrcoef(np.array([x[0:len(x)-t], x[t:len(x)]]))
+
+
+def _running_mean(inputData):
+    """
+    Calculates 5-point running mean for NINO index and the spectra
+
+    Author: Luke Van Roekel
+    Last Modified: 03/20/2017
+    """
+
+    return pd.Series.rolling(inputData, 5, center=True, min_periods=1).mean()
```

Comment:
Foiled again!

I have no objection.  Could you convert back to an xarray dataset right away (in `_running_mean`) instead of in the plotting routine?  That will mean minimal confusion with what is a pandas data set and what is xarray.